infrastructure/src/main/java/com/example/opencommerce/infra/catalog/ItemViewMapper.java

```diff
@@ -0,0 +1,66 @@
+package com.example.opencommerce.infra.catalog;
+
+import com.example.opencommerce.app.catalog.query.FullItemDto;
+import com.example.opencommerce.app.catalog.query.ImageDto;
+import com.example.opencommerce.app.catalog.query.ItemDto;
+import com.example.opencommerce.app.pricing.query.PriceDto;
+import com.example.opencommerce.app.pricing.query.PriceDto.DiscountDto;
+import com.example.opencommerce.infra.catalog.views.BaseItemView;
+import com.example.opencommerce.infra.catalog.views.FullItemView;
+import com.example.opencommerce.infra.catalog.views.ImageView;
+import com.example.opencommerce.infra.catalog.views.PriceView;
+import com.example.opencommerce.infra.catalog.views.PriceView.DiscountView;
+
+import java.util.List;
+
+import static java.util.Objects.isNull;
+
+class ItemViewMapper {
```

Comment:
Any reason for this to be a managed bean? Those methods are good candidates for a static factory methods. What is the difference between singleton managed by the framework and having a mapper with static methods?